Testsuite: Add test for monitoring feature

calancha opened the pull request:
## What does this PR change?

- Add a test for the monitoring feature.
(https://github.com/SUSE/spacewalk/issues/10991)

## Links
### Ports

Manager-4.0: https://github.com/SUSE/spacewalk/pull/11184
## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

srbarrios commented on testsuite/features/step_definitions/navigation_steps.rb:
```diff
@@ -876,3 +876,17 @@
+                when 'Prometheus postgres exporter' then [9187, "Postgres Exporter"]
+                else raise "Unknown port for service #{service}"
+                end
+  _output, code = node.run("curl -s http://localhost:#{port} | grep -i '#{text}'")
```
It might be better to assure that the endpoint can be requested outside localhost.

cavalheiro commented on testsuite/features/secondary/min_centos_monitoring.feature:
```diff
@@ -0,0 +1,60 @@
+  I want to enable Prometheus exporters
+
+  Scenario: Pre-requisite: enable Prometheus exporters repository on the CentOS minion
+    And I enable repository "tools_pool_repo" on this "ceos_ssh_minion"
```
Typo: 'cenos' -> 'centos'. 
Otherwise looks good to me!

Bischoff replied:
no, it's (ugly) conventional names

Bischoff commented on testsuite/features/secondary/min_centos_monitoring.feature:
```diff
@@ -0,0 +1,60 @@
+  I want to enable Prometheus exporters
+
+  Scenario: Pre-requisite: enable Prometheus exporters repository on the CentOS minion
+    And I enable repository "tools_pool_repo" on this "ceos_ssh_minion"
```
And -> When

Bischoff commented on testsuite/features/secondary/min_centos_monitoring.feature:
```diff
@@ -0,0 +1,60 @@
+    And I should see a "Enable and configure Prometheus exporters for managed systems." text
+    And I check "node_exporter#enabled"
+    And I check "apache_exporter#enabled"
+    And I check "postgres_exporter#enabled"
```
Ideally, we don't use internals like HTML or javascript internals. We use some clear English name, and convert it to the internal inside the step definition.

Look at we do for Retail formulas to have an example.

Bischoff commented on testsuite/features/secondary/min_centos_monitoring.feature:
```diff
@@ -0,0 +1,60 @@
+
+  Scenario: Apply highstate for Prometheus exporters on the CentOS minion
+     Given I am on the Systems overview page of this "ceos_ssh_minion"
+     And I follow "States" in the content area
```
When

Bischoff commented on testsuite/features/secondary/min_centos_monitoring.feature:
```diff
@@ -0,0 +1,60 @@
+     And I wait until event "Apply highstate scheduled by admin" is completed
+
+  Scenario: Visit monitoring endpoints on the CentOS minion
+    And I visit "Prometheus node exporter" endpoint of this "ceos_ssh_minion"
```
When

Bischoff commented on testsuite/features/secondary/min_centos_monitoring.feature:
```diff
@@ -0,0 +1,60 @@
+    Then I should see a "Formula saved" text
+
+  Scenario: Cleanup: apply highstate after test monitoring on the CentOS minion
+     Given I am on the Systems overview page of this "ceos_ssh_minion"
```
one less space (indentation)

Bischoff commented on testsuite/features/secondary/min_centos_monitoring.feature:
```diff
@@ -0,0 +1,60 @@
+    Then I should see a "Formula saved" text
+
+  Scenario: Apply highstate for Prometheus exporters on the CentOS minion
+     Given I am on the Systems overview page of this "ceos_ssh_minion"
```
One less space (indentation)

Bischoff commented on testsuite/features/secondary/min_monitoring.feature:
```diff
@@ -0,0 +1,73 @@
+  I want to enable Prometheus exporters
+
+  Scenario: Pre-requisite: enable Prometheus exporters repository on the minion
+    And I enable repository "tools_additional_repo" on this "sle_minion"
```
Same remarks as for CentOS one.

calancha commented:
@srbarrios In which YAML file should this monitoring test be included, `secondary.yml` or `secondary_parallelizable.yml`?

srbarrios commented:
> @srbarrios In which YAML file should this monitoring test be included, `secondary.yml` or `secondary_parallelizable.yml`?

If it doesnt have any dependency wirh others, then parallel.

calancha commented:
> Very cool - but look at requested changes.  Thank you.
I have addressed all your comments. 
I have also listed the added tests into `secondary_parallelizable.yml` (which I forgot yesterday).
Tested in my local environment with success.
